ui/app/controllers/jobs/run/templates/new.js

```diff
@@ -0,0 +1,71 @@
+import { A } from '@ember/array';
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+const ALL_NAMESPACE_WILDCARD = '*';
+
+export default class JobsRunTemplatesController extends Controller {
+  @service router;
+  @tracked templateName = null;
+  @tracked templateNamespace = 'default';
+
+  keyValues = A([]);
+
+  get namespaceOptions() {
+    const namespaces = this.store
+      .peekAll('namespace')
+      .map(({ name }) => ({ key: name, label: name }));
+
+    if (namespaces.length <= 1) return null;
+
+    // Create default namespace selection
+    namespaces.unshift({
+      key: ALL_NAMESPACE_WILDCARD,
+      label: 'All (*)',
+    });
+
+    return namespaces;
+  }
+
+  @action
+  updateKeyValue(key, value) {
+    if (this.keyValues.find((kv) => kv.key === key)) {
+      this.keyValues.find((kv) => kv.key === key).value = value;
+    } else {
+      this.keyValues.pushObject({ key, value });
+    }
+  }
+
+  @action
+  async save(e, overwrite = false) {
+    if (e.type === 'submit') {
+      e.preventDefault();
+    }
+
+    if (this.model?.isNew) {
+      if (this.namespaceOptions) {
+        this.model.set('namespace', this.templateNamespace);
+      } else {
+        const [namespace] = this.store.peekAll('namespace').toArray();
+        this.model.set('namespace', namespace.id);
+      }
+    }
+
+    this.model.set('keyValues', this.keyValues);
+    this.model.set('path', `nomad/job-templates/${this.templateName}`);
+    this.model.setAndTrimPath();
+    await this.model.save({ adapterOptions: { overwrite } });
+
+    this.flashMessages.add({
+      title: 'Job template saved',
+      message: `${this.templateName} successfully saved`,
+      type: 'success',
+      destroyOnClick: false,
+      timeout: 5000,
+    });
+
+    this.router.transitionTo('jobs.run.templates');
```

Comment:
@philrenaud there's some funky behavior for how we handle deleting a record on exiting the controller that has a slight conflict with saving. I handled the problem slightly differently than you did with variables by addressing this logic in `resetController`.

Reply:
Hm, I appreciate the different solution attempted here but I"m running into a breaking error when I try this:

`Error: Attempted to handle event `deleteRecord` on <variable:nomad/job-templates/my fun template@default> while in state root.loaded.created.inFlight. `

Reply:
I'm unable to reproduce this. Here's a [Replay](https://app.replay.io/recording/reproduction-of-error--ff185084-7610-4771-9474-92090d5bbd0d) showing that I'm getting to success.

Could you record a Replay so that I can trace what's happening to start debugging here?

Reply:
Look like calling `unloadRecord` patches our problem here.